src/Items/ItemEndCrystal.h

```diff
@@ -0,0 +1,83 @@
+
+#pragma once
+
+#include "ItemHandler.h"
+#include "../World.h"
+#include "../Entities/Player.h"
+
+
+
+
+
+class cItemEndCrystalHandler : public cItemHandler
+{
+	using Super = cItemHandler;
+
+  public:
+	cItemEndCrystalHandler(int a_ItemType) : Super(a_ItemType) {}
+
+
+
+
+
+	virtual bool OnItemUse(
+		cWorld * a_World, cPlayer * a_Player,
+		cBlockPluginInterface & a_PluginInterface, const cItem & a_HeldItem,
+		const Vector3i a_BlockPos,
+		eBlockFace a_ClickedBlockFace) override
+	{
+		// Must click a valid block:
+		if (a_ClickedBlockFace < 0)
+		{
+			return false;
+		}
+
+		auto * a_ChunkMap = a_World->GetChunkMap();
+
+		cChunkInterface ChunkInterface(a_ChunkMap);
+
+		// Don't place if two blocks above placement block aren't air
+		if (ChunkInterface.GetBlock(a_BlockPos + Vector3i(0, 1, 0)) != 0 ||
+			ChunkInterface.GetBlock(a_BlockPos + Vector3i(0, 2, 0)) != 0)
+		{
+			return false;
+		}
+
+		auto a_BlockType = ChunkInterface.GetBlock(a_BlockPos);
+
+		// When placed on obsidian or bedrock
+		if (a_BlockType == E_BLOCK_OBSIDIAN || a_BlockType == E_BLOCK_BEDROCK)
```

Comment:
Please use the early-return pattern here, like you did earlier with the other checks.